### PR TITLE
Some `plam` reductions

### PR DESCRIPTION
**Pull request by TotallyNotChase**
This should benefit #371 

I think there may be more places to optimize. Need to check.

### CODE REVIEW

File: Plutarch/Bool.hs [hunk 1]
```diff
@@ -111,7 +111,7 @@ pif b case_true case_false = pmatch b $ \case
 
 -- | Boolean negation for 'PBool' terms.
 pnot :: Term s (PBool :--> PBool)
-pnot = phoistAcyclic $ plam $ \x -> pif x (pcon PFalse) $ pcon PTrue
+pnot = phoistAcyclic $ plam $ \x -> pif' # x # pconstant False # pconstant True
```
Comment (L-as) [inline Plutarch/Bool.hs:114]: Why use `pconstant` when it's unnecessarily longer? Just use `pcon PFalse` and such

Comment (TotallyNotChase) [reply Plutarch/Bool.hs]: Habit, but yeah - changed.

**Comment by TotallyNotChase** [comment]
Seems like those were all the reductions I was able to spot.

**Review by L-as — APPROVED** [review]
LGTM

**Pull request merged**